File: Array/liuchang/day28/day28.java
public class day28 {

    public static int numOfSubarrays(int arr[], int arrSize, int k, int threshold) {
        int count = 0;
        for (int i = 0; i < arrSize - k + 1; i++) {
            int num = 0;
            for (int j = i; j < i+k; j++) {
                num += arr[j];
            }
            if (num / k >= threshold) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        int datas[] = {2, 2, 2, 2, 5, 5, 5, 8};
        System.out.println("=====" + numOfSubarrays(datas, 8, 3, 4));
    }
}

File: Array/liuchang/day28/day28.kt
package net.study.kotlintest

/**
 * @author       刘畅
 * @createdate   2020/6/28
 * @describe
 */
class day28 {

    fun numOfSubarrays(arr: IntArray, arrSize: Int, k: Int, threshold: Int): Int {
        var count = 0;
        for (i in 0 until (arrSize - k + 1)) {
            var num = 0;
            for (j in i until (i + k)) {
                num += arr[j];
            }
            if (num / k >= threshold) {
                count++;
            }
        }
        return count;
    }

    fun main(args: Array<String>) {
        var datas = intArrayOf(2, 2, 2, 2, 5, 5, 5, 8);
        System.out.println("=====" + numOfSubarrays(datas, 8, 3, 4));
    }
}